### PR TITLE
Making servers list an additional option for GlusterVolumeSource

### PR DESCRIPTION
**Pull request by humblec**
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/31616)

### CODE REVIEW

**Comment by rootfs** [comment]
@kubernetes/sig-storage @kubernetes/api-review-team

**Comment by rootfs** [comment]
this is a followup to https://github.com/openshift/origin/issues/6331

**Comment by erictune** [comment]
ok to test

File: pkg/api/types.go [hunk 1]
```diff
@@ -673,6 +673,9 @@ type GlusterfsVolumeSource struct {
 	// Required: EndpointsName is the endpoint name that details Glusterfs topology
 	EndpointsName string `json:"endpoints"`
 
+	// Optional: Servers is the gluster server list as a string seperated by a comma.
+	Servers string `json:"servers,omitempty"`
```
Comment (erictune) [inline pkg/api/types.go]: comma separate list is not common in our APIs.   Usually one sees `[]string`.  Is comma separate list the standard for specifying Glusterfs server lists?

Comment (humblec) [inline pkg/api/types.go]: @erictune Thanks for the mention ! In most of the cases, its just 'one' server like NFS . If at all, there are more than one, there is a tendency to mention it in comma separate list. Let me check with other folks what they think.  I wll act accordingly.

Comment (humblec) [inline pkg/api/types.go]: I think it make more sense to use []string regardless how gluster use it. I am updating the PR.. Thanks @erictune for your review!!

**Comment by erictune** [comment]
Please squash into one change with the types.go change, and one with all the autogenerated changes.  I will review after that.

**Comment by rootfs** [comment]
@humblec can you update README in glusterfs volume example? Rest LGTM.

**Comment by humblec** [comment]
@erictune and @rootfs addressed your comments in this update. Thanks !

**Comment by humblec** [comment]
I rebased the patch.  In last run of tests, the GCE Node e2e test fails with below error
`
command [ssh -i /home/jenkins/.ssh/google_compute_engine -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o CheckHostIP=no -o StrictHostKeyChecking=no -o ServerAliveInterval=30 -o LogLevel=ERROR 104.198.33.119 -- sh -c 'sudo mkdir -p /tmp/gcloud-e2e-411628582/cni ; sudo wget -O - https://storage.googleapis.com/kubernetes-release/network-plugins/cni-9d5e6e60e79491207834ae8439e80c943db65a69.tar.gz | sudo tar -xz -C /tmp/gcloud-e2e-411628582/cni'] failed with error: exit status 2 and output:
--2016-08-31 04:26:04--  https://storage.googleapis.com/kubernetes-release/network-plugins/cni-9d5e6e60e79491207834ae8439e80c943db65a69.tar.gz
Resolving storage.googleapis.com... 209.85.200.128, 2607:f8b0:4001:c0d::80
Connecting to storage.googleapis.com|209.85.200.128|:443... failed: Connection timed out.
Connecting to storage.googleapis.com|2607:f8b0:4001:c0d::80|:443... failed: Network is unreachable.

gzip: stdin: unexpected end of file
tar: Child returned status 1
tar: Error is not recoverable: exiting now
`
any thoughts on this error ?

File: pkg/api/types.go [hunk 2]
```diff
@@ -673,6 +673,9 @@ type GlusterfsVolumeSource struct {
 	// Required: EndpointsName is the endpoint name that details Glusterfs topology
 	EndpointsName string `json:"endpoints"`
 
+	// Optional: Servers is the gluster server list
```
Comment (erictune) [inline pkg/api/types.go]: Needs to say that this overrides EndpointsName.

File: pkg/api/types.go [hunk 3]
```diff
@@ -673,6 +673,9 @@ type GlusterfsVolumeSource struct {
 	// Required: EndpointsName is the endpoint name that details Glusterfs topology
```
Comment (erictune) [inline pkg/api/types.go]: Is this Required now?  

You cannot really make this un-Required because clients might be looking for this field?

**Comment by erictune** [comment]
I have concerns with this approach.  Commented on github.com/openshift/origin/issues/6331

**Comment by humblec** [comment]
@erictune thanks ! I have also shared my thoughts in above mentioned openshift PR .

**Comment by erictune** [comment]
Discussion ongoing on other issue.

**Pull request closed without merging**